pkgng module: declare BATCH mode before installing packages.

dlangille opened the pull request:
setting BATCH=yes means that the package will not ask any questions during the install process.  Case in point: mail/postfix
# /usr/sbin/pkg install -g -U -y mail/postfix

The following 1 packages will be installed:

```
    Installing postfix: 2.10.2,1
```

The installation will require 14 MB more space

0 B to be downloaded
Checking integrity... done
[1/1] Installing postfix-2.10.2,1...===> Creating users and/or groups.
Using existing group 'mail'.
Using existing group 'maildrop'.
Using existing group 'postfix'.
Using existing user 'postfix'.
Would you like to activate Postfix in /etc/mail/mailer.conf [n]?

dlangille committed:
declare BATCH mode before installing packages.

setting BATCH=yes means that the package will not ask any questions during the install process.  Case in point: mail/postfix

# /usr/sbin/pkg install -g -U -y mail/postfix
The following 1 packages will be installed:

        Installing postfix: 2.10.2,1

The installation will require 14 MB more space

0 B to be downloaded
Checking integrity... done
[1/1] Installing postfix-2.10.2,1...===> Creating users and/or groups.
Using existing group 'mail'.
Using existing group 'maildrop'.
Using existing group 'postfix'.
Using existing user 'postfix'.
Would you like to activate Postfix in /etc/mail/mailer.conf [n]?

mpdehaan commented:
Not sure why this would run "set" when it could just pass it as an environment variable when running the command?

BATCH=yes foo bar baz ...

seems like what this should do.

dlangille commented:
That's an even better idea.

dlangille committed:
declare BATCH mode before installing packages (better solution)

setting BATCH=yes means that the package will not ask any questions during the install process. Case in point: mail/postfix

dlangille committed:
remove previous commit

dlangille commented:
676eee3 is the one you want...

mpdehaan commented:
Apologies in the delay, in getting to this but we've had a large backlog -- also, since some time has passed, the underpinnings beneath this module have shifted.  We've gotten ourselves into a small conundrum here in that this would now require passing use_unsafe_shell=True to this module because of the environment, which we don't want to do as that would allow for command injection.  As such, we'll need to modify this to set the environment through Python.

mpdehaan commented:
Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

On September 26, 2014, due to enormous levels of contribution to the project Ansible decided to reorganize module repos, making it easier
for developers to work on the project and for us to more easily manage new contributions and tickets.

We split modules from the main project off into two repos, http://github.com/ansible/ansible-modules-core and http://github.com/ansible/ansible-modules-extras

If you still would like this pull request merged, we will need your help making this target the new repo.  If you do not take any action, this
pull request unfortunately cannot be applied.

We apologize that we are not able to make this transition happen seamlessly, though this is a one-time change and your help is greatly appreciated -- 
this will greatly improve velocity going forward.

Both sets of modules will ship with Ansible, though they'll receive slightly different ticket handling.

To locate where a module lives between 'core' and 'extras'
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your pull request to update the existing module to https://github.com/ansible/ansible-modules-core
- Note that python modules in ansible now also end in ".py" and this extension is required for new contributions.
- action_plugins (modules with server side components) still live in the main repo.  If your pull request touches both, which should be
   exceedingly rare, submit two new pull requests and make sure to mention the links to each other in the comments.

Otherwise, if this is a new module:
- Submit your pull request to add a module to https://github.com/ansible/ansible-modules-extras

It may be possible to re-patriate your pull requests automatically, one user-submitted approach for advanced git users
has been suggested at https://gist.github.com/willthames/afbaaab0c9681ed45619

Additionally, should you need more help with this, you can ask questions on:
- the development mailing list: https://groups.google.com/forum/#!forum/ansible-devel

Thank you very much!
